packages/qdrant-loader-mcp-server/src/qdrant_loader_mcp_server/search/hybrid_search.py

```
@@ -593,55 +640,517 @@ async def cluster_documents(
+    
+    def _analyze_cluster_relationships(self, clusters, documents: list[HybridSearchResult]) -> list[dict[str, Any]]:
+        """Analyze relationships between clusters."""
+        if len(clusters) < 2:
+            return []
+        
+        relationships = []
+        doc_lookup = {f"{doc.source_type}:{doc.source_title}": doc for doc in documents}
+        
+        # Analyze pairwise cluster relationships
+        for i, cluster_a in enumerate(clusters):
+            for j, cluster_b in enumerate(clusters[i+1:], i+1):
```

Ruff (0.12.2)

Loop control variable `j` not used within loop body

Rename unused `j` to `_j`

(B007)